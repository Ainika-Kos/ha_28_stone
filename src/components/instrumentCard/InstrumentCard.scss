@import "../../styles/variables";

.instrumentCard {
    width: 210px;
    margin: 10px;
    padding: 5px;
    background-color: $cardBackgroundColor;
    border-radius: 4px;
    box-shadow: 0 0.1875rem 1.5rem $mainShadow;
}

.instrumentCard__currency {
    text-align: center;
    font-weight: 500;
    padding: 8px;
    border-radius: 4px;
    background-color: $currencyBackgroundColor;
    & p {
        margin: 0;
    }
}

.instrumentCard__deal-info {
    display: flex;
    justify-content: center;
    margin: 5px;
    border-radius: 4px;
}

.instrumentCard__side {
    padding: 3px 5px;
    border-radius: 4px;
}

.instrumentCard__amount {
    font-size: 10px;
    font-weight: 700;
    vertical-align: sub;
    margin: 0 2px;
}

.instrumentCard__price {
    font-size: 20px;
    &.bid {
        color: $blueTextColor;
    }
    &.offer {
        color: $purpleTextColor;
    }

}