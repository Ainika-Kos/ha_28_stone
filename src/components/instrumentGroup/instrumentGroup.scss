.instrumentGroup {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
}

.instrumentGroup__title {
    display: flex;
    align-items: center;
    justify-content: center;
    margin-bottom: 10px;
}

.instrumentGroup__image {
    width: 30px;
    height: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
}

.instrumentGroup__title__text {
    margin: 0;
    margin-left: 20px;
    font-size: 20px;
    font-weight: 500;
}