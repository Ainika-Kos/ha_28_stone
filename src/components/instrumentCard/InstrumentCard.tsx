import React, { FC } from 'react';
import { InstrumentType } from '../../types/types';
import './InstrumentCard.scss';

export const InstrumentCard: FC<InstrumentType> =
  ({ currencyPair, bidPrice, bidAmount, offerPrice, offerAmount })  => {
  
    return (
      <div className="instrumentCard">
        <div className="instrumentCard__currency">
          <p>{currencyPair}</p>
        </div>
        <div className="instrumentCard__deal-info">
          <div className="instrumentCard__side">
            <span className="instrumentCard__amount">{bidAmount}</span>
            <span className="instrumentCard__price bid">{bidPrice}</span>
          </div>
          <div className="instrumentCard__side">
            <span className="instrumentCard__price offer">{offerPrice}</span>
            <span className="instrumentCard__amount">{offerAmount}</span>
          </div>
        </div>
      </div>
    );
  };