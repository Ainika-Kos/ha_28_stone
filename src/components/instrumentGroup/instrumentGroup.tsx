import React, { FC } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { InstrumentType } from '../../data/types';
import { InstrumentCard } from '../instrumentCard/InstrumentCard';
import './instrumentGroup.scss';

type InstrumentGroupType = {
  instrumentArray: InstrumentType[];
  currency: string;
  image: string;
};

export const InstrumentGroup: FC<InstrumentGroupType> =
    ({
      instrumentArray,
      currency,
      image,
    }) => {

    
      return (
        <div className="instrumentGroup">
          <div className="instrumentGroup__title">
            <img src={image} className="instrumentGroup__image" alt={currency} />
            <p className="instrumentGroup__title__text">{currency}</p>
          </div>
          {instrumentArray.map(({
            currencyPair,
            bidPrice,
            bidAmount,
            offerPrice,
            offerAmount
          }) => {
            return (
              <div key={uuidv4()}>
                <InstrumentCard
                  currencyPair={currencyPair}
                  bidPrice={bidPrice}
                  bidAmount={bidAmount}
                  offerPrice={offerPrice}
                  offerAmount={offerAmount}
                />
              </div>
            );
          })}
        </div>
      );
    };