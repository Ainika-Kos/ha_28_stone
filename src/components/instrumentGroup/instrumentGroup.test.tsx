import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { InstrumentGroup } from './instrumentGroup';
import { InstrumentCard } from '../instrumentCard/InstrumentCard';

Enzyme.configure({ adapter: new Adapter() });

const currencies = [{
  'currencyPair': 'ETH | CHF',
  'bidPrice': 225.57,
  'bidAmount': 10,
  'offerPrice': 235.42,
  'offerAmount': 30
},
];
const currency = 'Ethereum';
const image = 'Image source 2';
    

describe('Rendering InstrumentGroup component without crashing', () => {
  const wrapper = shallow(
    <InstrumentGroup
      instrumentArray={currencies}
      currency={currency[0]}
      image={image}
    />
  );
    
  it('renders <InstrumentGroup /> component', () => {
    expect(wrapper).toHaveLength(1);
  });
  it('renders <InstrumentCard /> component', () => {
    expect(wrapper.find('InstrumentCard')).toHaveLength(1);
  });
  
});

describe('Passing props to InstrumentCard', () => {

  const wrapper = shallow(
    <InstrumentCard
      currencyPair={currencies[0].currencyPair}
      bidPrice={currencies[0].bidPrice}
      bidAmount={currencies[0].bidAmount}
      offerPrice={currencies[0].offerPrice}
      offerAmount={currencies[0].offerAmount}
    />
  );
      
  it('accepts currency name prop', () => {
    expect(wrapper.contains(<p>ETH | CHF</p>)).toBeTruthy();
  });
       
  it('accepts currency image prop', () => {
    expect(wrapper.contains(<span className="instrumentCard__amount">{currencies[0].bidAmount}</span>)).toBeTruthy();
  });
      
  it('accepts currency image prop', () => {
    expect(wrapper.contains(<span className="instrumentCard__price bid">{currencies[0].bidPrice}</span>)).toBeTruthy();
  });
    
  it('accepts currency image prop', () => {
    expect(wrapper.contains(<span className="instrumentCard__price offer">{currencies[0].offerPrice}</span>)).toBeTruthy();
  });
    
  it('accepts currency image prop', () => {
    expect(wrapper.contains(<span className="instrumentCard__amount">{currencies[0].offerAmount}</span>)).toBeTruthy();
  });
});

  