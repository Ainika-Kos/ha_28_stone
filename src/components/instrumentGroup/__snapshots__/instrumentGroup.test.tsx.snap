// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="instrumentCard"
>
  <div
    className="instrumentCard__currency"
  >
    <p />
  </div>
  <div
    className="instrumentCard__deal-info"
  >
    <div
      className="instrumentCard__side"
    >
      <span
        className="instrumentCard__amount"
      />
      <span
        className="instrumentCard__price bid"
      />
    </div>
    <div
      className="instrumentCard__side"
    >
      <span
        className="instrumentCard__price offer"
      />
      <span
        className="instrumentCard__amount"
      />
    </div>
  </div>
</div>
`;
